<html><head><title>Estimation and tests of hierarchical F-statistics</title>
<link rel="stylesheet" type="text/css" href="Rchm.css">
</head><body>
<h1>Estimation and tests of hierarchical F-statistics
<img class="toplogo" src="logo.jpg" alt="[R logo]"></h1>

<hr>

<object type="application/x-oleobject" classid="clsid:1e2a7bd0-dab9-11d0-b93a-00c04fc99f9e">
<param name="keyword" value=".. contents">
</object>

<h2>Help pages for package `hierfstat' version 0.04-4</h2>


<table width="100%">
<tr><td width="25%"><a href="boot.vc.html">boot.vc</a></td>
<td>Bootstrap confidence intervals for variance components</td></tr>
<tr><td width="25%"><a href="exhier.html">exhier</a></td>
<td>Example data set with 4 levels, one diploid and one haploid locus</td></tr>
<tr><td width="25%"><a href="g.stats.html">g.stats</a></td>
<td>Calculates likelihood-ratio G-statistic on contingency table</td></tr>
<tr><td width="25%"><a href="g.stats.glob.html">g.stats.glob</a></td>
<td>Likelihood ratio G-statistic over loci</td></tr>
<tr><td width="25%"><a href="genot2al.html">genot2al</a></td>
<td>Separates diploid genotypes in its constituant alleles</td></tr>
<tr><td width="25%"><a href="gtrunchier.html">gtrunchier</a></td>
<td>Genotypes at 6 microsatellite loci of Galba truncatula from different patches in Western Switzerland</td></tr>
<tr><td width="25%"><a href="hierfstat.html">hierfstat</a></td>
<td>General information on the hierfstat package</td></tr>
<tr><td width="25%"><a href="prepdata.html">prepdata</a></td>
<td>Sort and renumber and recode levels for the hierarchical analysis if necessary</td></tr>
<tr><td width="25%"><a href="read.fstat.data.html">read.fstat.data</a></td>
<td>Reads data from a FSTAT file</td></tr>
<tr><td width="25%"><a href="samp.between.html">samp.between</a></td>
<td>Shuffles a sequence among groups defined by the input vector</td></tr>
<tr><td width="25%"><a href="samp.between.within.html">samp.between.within</a></td>
<td>Shuffles a sequence</td></tr>
<tr><td width="25%"><a href="samp.within.html">samp.within</a></td>
<td>Shuffles a sequence within groups defined by the input vector</td></tr>
<tr><td width="25%"><a href="test.between.html">test.between</a></td>
<td>Tests the significance of the effect of test.lev on genetic differentiation</td></tr>
<tr><td width="25%"><a href="test.between.within.html">test.between.within</a></td>
<td>Tests the significance of the effect of test.lev on genetic differentiation</td></tr>
<tr><td width="25%"><a href="test.g.html">test.g</a></td>
<td>Tests the significance of the effect of level on genetic differentiation</td></tr>
<tr><td width="25%"><a href="test.within.html">test.within</a></td>
<td>Tests the significance of the effect of inner.level on genetic differentiation within blocks defined by outer.level</td></tr>
<tr><td width="25%"><a href="varcomp.html">varcomp</a></td>
<td>Estimates variance components for each allele of a locus</td></tr>
<tr><td width="25%"><a href="varcomp.glob.html">varcomp.glob</a></td>
<td>Estimate variance components and hierarchical F-statistics over all loci</td></tr>
<tr><td width="25%"><a href="yangex.html">yangex</a></td>
<td>Example data set from Yang (1998) appendix</td></tr>
</table>
</body></html>
