<html><head><title>Reads data from a FSTAT file</title>
<link rel="stylesheet" type="text/css" href="Rchm.css">
</head>
<body>

<table width="100%"><tr><td>read.fstat.data(hierfstat)</td><td align="right">R Documentation</td></tr></table><object type="application/x-oleobject" classid="clsid:1e2a7bd0-dab9-11d0-b93a-00c04fc99f9e">
<param name="keyword" value="R:   read.fstat.data">
<param name="keyword" value=" Reads data from a FSTAT file">
</object>


<h2>Reads data from a FSTAT file</h2>


<h3>Description</h3>

<p>
Imports a <EM>FSTAT</EM> data file into R, skipping the unecessary information. The data frame created is made of <KBD>nl+1</KBD> columns, <KBD>nl</KBD> being the number of loci. The first column corresponds to the Population identifier, the following columns contains the genotypes of the individuals.
</p>


<h3>Usage</h3>

<pre>
read.fstat.data(filename,nloc,na.s="0")
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>filename</code></td>
<td>
a file in the FSTAT format (<a href="http://www.unil.ch/popgen/softwares/fstat.htm">http://www.unil.ch/popgen/softwares/fstat.htm</a>):  The file must have the following format:
<br>
The first line contains 4 numbers: the number of samples, <KBD>np</KBD> , the number of loci, <KBD>nl</KBD>, the highest number used to label an allele, <KBD>nu</KBD>, and a 1 if the code for alleles is a one digit number (1-9), a 2 if code for alleles is a 2 digit number (01-99) or a 3 if code for alleles is a 3 digit number (001-999).  These 4 numbers need to be separated by any number of spaces. <EM>Actually, the function read.fstat.dat skips the first line. The info is just here to make sure that if you were to run the data set in fstat, it will work.</EM>
<br>
The first line is immediately followed by <KBD>nl</KBD> lines, each containing the name of a locus, in the order they will appear in the rest of the file.
<br>
On line <KBD>nl+2</KBD>, a series of numbers as follow:
<br>
<pre>1     0102   0103   0101  0203          0      0303</pre>
<br>
The first number identifies the sample to which the individual belongs, the second is the genotype of the individual at the first locus, coded with a 2 digits number for each allele, the third is the genotype at the second locus, until locus nl is entered (in the example above, <KBD>nl=6</KBD>).  Missing genotypes are encoded with 0.  Note that 0001 or 0100 are not a valid format, that is, both alleles at a locus have to be known, otherwise, the genotype is considered as missing.
No empty lines are needed between samples.  </td></tr>
<tr valign="top"><td><code>nloc</code></td>
<td>
The number of loci</td></tr>
<tr valign="top"><td><code>na.s</code></td>
<td>
The string that corresponds to the missing value. <EM>You should note have to change this</EM></td></tr>
</table>

<h3>Value</h3>

<p>
a data frame containing the desired data, in a format adequate to pass to <KBD>varcomp</KBD></p>

<h3>References</h3>

<p>
Goudet J. (1995). FSTAT (Version 1.2): A computer program to calculate F- statistics. Journal of Heredity 86:485-486
</p>
<p>
Goudet J. (2005). Hierfstat, a package for R to compute and test variance components and F-statistics. Molecular Ecology Notes. 5:184-186
</p>


<h3>Examples</h3>

<pre>
read.fstat.data(paste(.path.package("hierfstat"),"/data/diploid.dat",sep="",collapse=""),nloc=5)
</pre>



<hr><div align="center">[Package <em>hierfstat</em> version 0.03-1 <a href="00Index.html">Index]</a></div>

</body></html>
