<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Jerome Goudet" />

<meta name="date" content="2016-08-16" />

<title>Hierfstat latest features</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; }
code > span.dt { color: #902000; }
code > span.dv { color: #40a070; }
code > span.bn { color: #40a070; }
code > span.fl { color: #40a070; }
code > span.ch { color: #4070a0; }
code > span.st { color: #4070a0; }
code > span.co { color: #60a0b0; font-style: italic; }
code > span.ot { color: #007020; }
code > span.al { color: #ff0000; font-weight: bold; }
code > span.fu { color: #06287e; }
code > span.er { color: #ff0000; font-weight: bold; }
</style>



<link href="data:text/css,body%20%7B%0A%20%20background%2Dcolor%3A%20%23fff%3B%0A%20%20margin%3A%201em%20auto%3B%0A%20%20max%2Dwidth%3A%20700px%3B%0A%20%20overflow%3A%20visible%3B%0A%20%20padding%2Dleft%3A%202em%3B%0A%20%20padding%2Dright%3A%202em%3B%0A%20%20font%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0A%20%20font%2Dsize%3A%2014px%3B%0A%20%20line%2Dheight%3A%201%2E35%3B%0A%7D%0A%0A%23header%20%7B%0A%20%20text%2Dalign%3A%20center%3B%0A%7D%0A%0A%23TOC%20%7B%0A%20%20clear%3A%20both%3B%0A%20%20margin%3A%200%200%2010px%2010px%3B%0A%20%20padding%3A%204px%3B%0A%20%20width%3A%20400px%3B%0A%20%20border%3A%201px%20solid%20%23CCCCCC%3B%0A%20%20border%2Dradius%3A%205px%3B%0A%0A%20%20background%2Dcolor%3A%20%23f6f6f6%3B%0A%20%20font%2Dsize%3A%2013px%3B%0A%20%20line%2Dheight%3A%201%2E3%3B%0A%7D%0A%20%20%23TOC%20%2Etoctitle%20%7B%0A%20%20%20%20font%2Dweight%3A%20bold%3B%0A%20%20%20%20font%2Dsize%3A%2015px%3B%0A%20%20%20%20margin%2Dleft%3A%205px%3B%0A%20%20%7D%0A%0A%20%20%23TOC%20ul%20%7B%0A%20%20%20%20padding%2Dleft%3A%2040px%3B%0A%20%20%20%20margin%2Dleft%3A%20%2D1%2E5em%3B%0A%20%20%20%20margin%2Dtop%3A%205px%3B%0A%20%20%20%20margin%2Dbottom%3A%205px%3B%0A%20%20%7D%0A%20%20%23TOC%20ul%20ul%20%7B%0A%20%20%20%20margin%2Dleft%3A%20%2D2em%3B%0A%20%20%7D%0A%20%20%23TOC%20li%20%7B%0A%20%20%20%20line%2Dheight%3A%2016px%3B%0A%20%20%7D%0A%0Atable%20%7B%0A%20%20margin%3A%201em%20auto%3B%0A%20%20border%2Dwidth%3A%201px%3B%0A%20%20border%2Dcolor%3A%20%23DDDDDD%3B%0A%20%20border%2Dstyle%3A%20outset%3B%0A%20%20border%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0A%20%20border%2Dwidth%3A%202px%3B%0A%20%20padding%3A%205px%3B%0A%20%20border%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0A%20%20border%2Dwidth%3A%201px%3B%0A%20%20border%2Dstyle%3A%20inset%3B%0A%20%20line%2Dheight%3A%2018px%3B%0A%20%20padding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0A%20%20border%2Dleft%2Dstyle%3A%20none%3B%0A%20%20border%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0A%20%20background%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0A%0Ap%20%7B%0A%20%20margin%3A%200%2E5em%200%3B%0A%7D%0A%0Ablockquote%20%7B%0A%20%20background%2Dcolor%3A%20%23f6f6f6%3B%0A%20%20padding%3A%200%2E25em%200%2E75em%3B%0A%7D%0A%0Ahr%20%7B%0A%20%20border%2Dstyle%3A%20solid%3B%0A%20%20border%3A%20none%3B%0A%20%20border%2Dtop%3A%201px%20solid%20%23777%3B%0A%20%20margin%3A%2028px%200%3B%0A%7D%0A%0Adl%20%7B%0A%20%20margin%2Dleft%3A%200%3B%0A%7D%0A%20%20dl%20dd%20%7B%0A%20%20%20%20margin%2Dbottom%3A%2013px%3B%0A%20%20%20%20margin%2Dleft%3A%2013px%3B%0A%20%20%7D%0A%20%20dl%20dt%20%7B%0A%20%20%20%20font%2Dweight%3A%20bold%3B%0A%20%20%7D%0A%0Aul%20%7B%0A%20%20margin%2Dtop%3A%200%3B%0A%7D%0A%20%20ul%20li%20%7B%0A%20%20%20%20list%2Dstyle%3A%20circle%20outside%3B%0A%20%20%7D%0A%20%20ul%20ul%20%7B%0A%20%20%20%20margin%2Dbottom%3A%200%3B%0A%20%20%7D%0A%0Apre%2C%20code%20%7B%0A%20%20background%2Dcolor%3A%20%23f7f7f7%3B%0A%20%20border%2Dradius%3A%203px%3B%0A%20%20color%3A%20%23333%3B%0A%20%20white%2Dspace%3A%20pre%2Dwrap%3B%20%20%20%20%2F%2A%20Wrap%20long%20lines%20%2A%2F%0A%7D%0Apre%20%7B%0A%20%20border%2Dradius%3A%203px%3B%0A%20%20margin%3A%205px%200px%2010px%200px%3B%0A%20%20padding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0A%20%20background%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0A%0Acode%20%7B%0A%20%20font%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0A%20%20font%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0A%20%20padding%3A%202px%200px%3B%0A%7D%0A%0Adiv%2Efigure%20%7B%0A%20%20text%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0A%20%20background%2Dcolor%3A%20%23FFFFFF%3B%0A%20%20padding%3A%202px%3B%0A%20%20border%3A%201px%20solid%20%23DDDDDD%3B%0A%20%20border%2Dradius%3A%203px%3B%0A%20%20border%3A%201px%20solid%20%23CCCCCC%3B%0A%20%20margin%3A%200%205px%3B%0A%7D%0A%0Ah1%20%7B%0A%20%20margin%2Dtop%3A%200%3B%0A%20%20font%2Dsize%3A%2035px%3B%0A%20%20line%2Dheight%3A%2040px%3B%0A%7D%0A%0Ah2%20%7B%0A%20%20border%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0A%20%20padding%2Dtop%3A%2010px%3B%0A%20%20padding%2Dbottom%3A%202px%3B%0A%20%20font%2Dsize%3A%20145%25%3B%0A%7D%0A%0Ah3%20%7B%0A%20%20border%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0A%20%20padding%2Dtop%3A%2010px%3B%0A%20%20font%2Dsize%3A%20120%25%3B%0A%7D%0A%0Ah4%20%7B%0A%20%20border%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0A%20%20margin%2Dleft%3A%208px%3B%0A%20%20font%2Dsize%3A%20105%25%3B%0A%7D%0A%0Ah5%2C%20h6%20%7B%0A%20%20border%2Dbottom%3A%201px%20solid%20%23ccc%3B%0A%20%20font%2Dsize%3A%20105%25%3B%0A%7D%0A%0Aa%20%7B%0A%20%20color%3A%20%230033dd%3B%0A%20%20text%2Ddecoration%3A%20none%3B%0A%7D%0A%20%20a%3Ahover%20%7B%0A%20%20%20%20color%3A%20%236666ff%3B%20%7D%0A%20%20a%3Avisited%20%7B%0A%20%20%20%20color%3A%20%23800080%3B%20%7D%0A%20%20a%3Avisited%3Ahover%20%7B%0A%20%20%20%20color%3A%20%23BB00BB%3B%20%7D%0A%20%20a%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0A%20%20%20%20text%2Ddecoration%3A%20underline%3B%20%7D%0A%20%20a%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0A%20%20%20%20text%2Ddecoration%3A%20underline%3B%20%7D%0A%0A%2F%2A%20Class%20described%20in%20https%3A%2F%2Fbenjeffrey%2Ecom%2Fposts%2Fpandoc%2Dsyntax%2Dhighlighting%2Dcss%0A%20%20%20Colours%20from%20https%3A%2F%2Fgist%2Egithub%2Ecom%2Frobsimmons%2F1172277%20%2A%2F%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%2F%2A%20Keyword%20%2A%2F%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%2F%2A%20DataType%20%2A%2F%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%2F%2A%20DecVal%20%28decimal%20values%29%20%2A%2F%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%2F%2A%20BaseN%20%2A%2F%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%2F%2A%20Float%20%2A%2F%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%2F%2A%20Char%20%2A%2F%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%2F%2A%20String%20%2A%2F%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%2F%2A%20Comment%20%2A%2F%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%2F%2A%20OtherToken%20%2A%2F%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%2F%2A%20AlertToken%20%2A%2F%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%2F%2A%20Function%20calls%20%2A%2F%20%0Acode%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%2F%2A%20ErrorTok%20%2A%2F%0A%0A" rel="stylesheet" type="text/css" />

</head>

<body>




<h1 class="title toc-ignore">Hierfstat latest features</h1>
<h4 class="author"><em>Jerome Goudet</em></h4>
<h4 class="date"><em>2016-08-16</em></h4>



<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>This vignette documents the latest developments in <code>hierfstat</code>. Refer to the other documents for an introduction to the package</p>
</div>
<div id="loading-data" class="section level1">
<h1>Loading data</h1>
<p><code>hierfstat</code> can now read <code>genind</code> objects (from package <code>adegenet</code>). Note however that only some genetic data types will be properly converted and used. The alleles need to be encoded either as integer (up to three digits per allele), or as nucleotides (<code>c(&quot;a&quot;,&quot;c&quot;,&quot;g&quot;,&quot;t&quot;,&quot;A&quot;,&quot;C&quot;,&quot;G&quot;,&quot;T&quot;)</code>).</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(adegenet)
<span class="kw">library</span>(hierfstat)</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(nancycats) 
<span class="kw">is.genind</span>(nancycats)</code></pre>
<pre><code>## [1] TRUE</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">boxplot</span>(<span class="kw">basic.stats</span>(nancycats)$perloc[,<span class="dv">1</span>:<span class="dv">3</span>]) <span class="co"># boxplot of Ho, Hs, Ht</span></code></pre>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAAABlBMVEUAAAD///+l2Z/dAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAEI0lEQVR4nO3a23LbOBAAUer/fzqxk5RlyWHjNsCQ6H7IuirEQD4rg7KU42GnHasfQPYEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggYDHZdpFdDYcXEJBAkE7QlUcdrsCfQUPRCBOv++/sIl49oTCBKoM4EggaDtgTyDIIEggSCBIIGe9mx5l2wroJYrBYIrBYIrJwPVvxc+MM8g2lMg2FOgUXveCoieAi173gqofKhAAvUNFYiAvIsFJBAkEHQroIihVwVqOnBDH2jVPnzx7kD4sHYH+vgOxo3rWp8V6M//5DHjutbnBfr84n9GAo0c17VeoNidLvs6aNZOuYDK79P3AWo9pCPe22kp7edimwKVjxOo74LLA9FPUpa7GP6wBwEdb190jetZf3rlgaPCXygW3MVafjEfA3TwrBRA48bCTj//3T2BhrwftAyo7gwa9hjqD7ZVZ1DdXWwh0LK7WNW4VY9h7J4CjbqwfpxAMG7IDSngMdbNFGju5gJVjFt5m+/eM2rzcUBNz7Xyrg8UXAqgUWdQRBmAZuzX3O5A+LTcHGjdb/Pd46YArXs/aPa4xvIBLb0hvZcPKGpca55BuI13sc4Egu4EFHL05wLq+g2MD9yWcgH1Txw+V6CisSMvXDLuIVDZSM+g05n3v4slTCBIICgG6O8d5eRQEOj5j95xK4sDOk7XCSQQXSzQ2cW/T+fzl/6bAz3+vaz1Lha+7+oEgsKBPKTrtsv1wWFB/ohBAkFRQPSTtDvQ8fZF17iFRb7dcbZOIFgnEKzbHMgziC/vvIuleSmZ9HVQzIeALeUECvoYuSWBIIGgnECeQd7FJo+LaxnQZVoEFLLfrDWxgwL3EyjJmthBgfsJlGRN7KDA/QRKsiZ2UOB+AiVZEzvorgkECQQJBAkECQQJBAkECQQJBAkECQQJBE0COl7+G7WmdRXPC24J0JDvTaDCecF9+2YLP9d8XlP+WejXqprPT3lecN++2cJtX58LZQ/1qs+gr4/Ei/+R0Nuatp06m/8MKgd6XdO2U2fXAGo6g7YCKlzy006dCVQ4LzjvYoXbzHwdNOYfRPrLKiQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBD0C9lKPUPJPAowAAAAAElFTkSuQmCC" /><!-- --></p>
</div>
<div id="descriptive-statistics" class="section level1">
<h1>Descriptive statistics</h1>
<p>The function you are the most likely to want using is <code>basic.stats</code> (it calculates <span class="math">\(H_O\)</span>, <span class="math">\(H_S\)</span>, <span class="math">\(F_{IS}\)</span>, <span class="math">\(F_{ST}\)</span> etc…). You can also get e.g. <code>allele.count</code> and <code>allelic.richness</code>, a rarefied measure of the number of alleles at each locus and in each population. For instance, below is a boxplot of the allelic richness for the 5 first loci in the nancycats dataset</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">boxplot</span>(<span class="kw">t</span>(<span class="kw">allelic.richness</span>(nancycats)$Ar[<span class="dv">1</span>:<span class="dv">5</span>,])) <span class="co">#5 first loci</span></code></pre>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAAABlBMVEUAAAD///+l2Z/dAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAFBElEQVR4nO3cgXKiOhhAYXz/l95dt7RqSU6SP4EA58zsnWkxCX4iWGvv8rBsy9E7MHsCQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQZ2BltN0FNDmN8t3Z7dmAlr6LxNvIqBlxDrhBIIEgiYC8hyE03kVGzbduASCBIIEggSCBILGAPGN7w6EL2juDvTPqN90RzbwKbb1blP9+1BHN/gclJQQqOd0RyYQ5OsgSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIGgWoKaPBVYumZ84sXkWoJJFYnTw4ZrU5p2B4CjJP8ShvYCPZyU3n+cIgntYtqBArcNPAsQjYs+x+c9Bkeli56BrXMVgzBG/j50LaMKXj/cACrwMvQdQYHaBeg26MlDuD8TmAhobrBl8zC4ABAkEnQHoULZTA8Hf2Zb9Ge71r2K0D8HhsUFVs3/95F30aFb8QfbFgH7+k52u4hkxFmjPc9DLOxOJcQJtA208f/qeUyvj5/epjqD+8eKDgP4+Avm3SG8O9Fjf/iu5ipWv2P8cdCBQ8XQCwXQCwXTHAh10FauZTiCY7pavg2qmg32MbW6+VycCys8QPP4Eom/hATgBUP4mY4F488WB0kdI6TVyfiB4BjY+QYtfRAhUvKO9blg7nUAw3dhzUMEO5Ls6EBxh27vE343csHY6gd5m+b1/wXNQ6+hJgTami/2w0PEcdAGgmlfHNUC5Q2sCoNCT4AZXsdigmwHV/zgpEGwWqGmeh0DfIxpfJt0GCIZfDSg4SKDq7QLB9qsB1V/mYXvND6tta9bfMDJd+qLdct+6dBKg4zoT0M7Hztei3W/4Oqb8U+8FT6EF9yJ4CosNqpv9eTdz96j1h4rsuBBQ4oEZA/Skyd6j6YBSj+Y4oM17FLgSjQVKzr4zUNt0P0Maz0F8htsZ6OfWPf+P5COvYnsDrTdP3qXpXvHsew7afboO7XoV2326cZMK1GtOgfZdXKBdpxv4bsg1gAYmECQQJBB0GNBpOgioaZH632r0nD22eJ8Eii0ikECxRQQSKLaIQALFFhHozAkECQQJBAkECQQJBAkECQQJBAkECQQJBI0CSnxsqeDXLfDpNZhia+v3L3oqf+Pzsmr3tqdNfR6ORz7Wuw5TpDctfJPKGUNlPvbW+PbNywfYLwD0dSCvh/P6ZcGK2yOfowruYmp0u8/QI2h5+ff+RfXIV6DcSSQxek6g9wfv/97yGXJ75MsBmJ0isW67zx5Az6+XZeO7xSPfTz1wBP1ed3ag9T4Wn6Q/R34OzJ+Kf637mBzo4yxbAbQO+PzIShnQaZ5icaDPLdVA7T57XMW+Ts+P992vG/n9TZgiMXpSoPUU+Xx21FzFNkZ+bqlc9317TYOArpNAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkEDQH7geO8S0RLyBAAAAAElFTkSuQmCC" /><!-- --></p>
</div>
<div id="populations" class="section level1">
<h1>Populations</h1>
<p>Population statistics are obtained through <code>basic.stats</code> or <code>wc</code> (<code>varcomp.glob</code> can also be used and will give the same result as <code>wc</code> for a one level hierarchy). For instance, <span class="math">\(F_{IS}\)</span> and <span class="math">\(F_{ST}\)</span> in the <em>Galba truncatula</em> dataset provided with <code>hierfstat</code> are obtained as:</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(gtrunchier) 
<span class="kw">wc</span>(gtrunchier[,-<span class="dv">2</span>])</code></pre>
<pre><code>## $FST
## [1] 0.540894
## 
## $FIS
## [1] 0.8154694</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">varcomp.glob</span>(<span class="kw">data.frame</span>(gtrunchier[,<span class="dv">1</span>]),gtrunchier[,-<span class="kw">c</span>(<span class="dv">1</span>:<span class="dv">2</span>)])$F <span class="co">#same</span></code></pre>
<pre><code>##                 gtrunchier...1.       Ind
## Total                  0.540894 0.9152809
## gtrunchier...1.        0.000000 0.8154694</code></pre>
<p>Confidence intervals on these statistics can be obtained via <code>boot.vc</code>:</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">boot.vc</span>(gtrunchier[,<span class="dv">1</span>],gtrunchier[,-<span class="kw">c</span>(<span class="dv">1</span>:<span class="dv">2</span>)])$ci</code></pre>
<pre><code>##       H-Total F-Pop/Total F-Ind/Total  H-Pop F-Ind/Pop   Hobs
## 2.5%   0.6573      0.4808      0.9046 0.2673    0.7724 0.0557
## 50%    0.7484      0.5406      0.9153 0.3416    0.8155 0.0630
## 97.5%  0.8129      0.6135      0.9263 0.4103    0.8492 0.0700</code></pre>
<p><code>boot.ppfis</code>and <code>boot.ppfst</code> provide bootsrap confidence intervals (bootstrapping over loci) for population specific <span class="math">\(F_{IS}\)</span> and pairwise <span class="math">\(F_{ST}\)</span> respectively.</p>
</div>
<div id="genetic-distances" class="section level1">
<h1>Genetic distances</h1>
<p><code>genet.dist</code> estimates one of 10 different genetic distances between populations as described mostly in Takezaki &amp; Nei (1996)</p>
<pre class="sourceCode r"><code class="sourceCode r">(Ds&lt;-<span class="kw">genet.dist</span>(gtrunchier[,-<span class="dv">2</span>],<span class="dt">method=</span><span class="st">&quot;Ds&quot;</span>)) <span class="co"># Nei's standard genetic distances</span></code></pre>
<pre><code>##           1         2         3         4         5
## 2 0.4272210                                        
## 3 1.1402899 1.8235430                              
## 4 0.8387367 0.9540338 1.6638485                    
## 5 0.6967425 0.6205417 2.5798363 0.8767008          
## 6 0.9411656 0.9742812 1.1553423 0.5243353 1.1911894</code></pre>
</div>
<div id="population-principal-coordinate-analysis" class="section level1">
<h1>Population Principal coordinate analysis</h1>
<p>Principal coordinate analysis can be carried out on this genetic distances:</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">pcoa</span>(<span class="kw">as.matrix</span>(Ds))</code></pre>
</div>
<div id="individual-pca" class="section level1">
<h1>Individual PCA</h1>
<p><code>hierfstat</code> has a function <code>indpca</code> carrying out Principal component analysis on individual genotypes</p>
<pre class="sourceCode r"><code class="sourceCode r">x&lt;-<span class="kw">indpca</span>(gtrunchier[,-<span class="dv">2</span>],<span class="dt">ind.labels=</span>gtrunchier[,<span class="dv">2</span>])
<span class="kw">plot</span>(x,<span class="dt">col=</span>gtrunchier[,<span class="dv">1</span>],<span class="dt">cex=</span><span class="fl">0.7</span>)</code></pre>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAAAFVBMVEUAAAAAAP8AzQAA////AAD/AP/////iDJLhAAAACXBIWXMAAA7DAAAOwwHHb6hkAAALuUlEQVR4nO2ciZqcKhSEsW15/0eOsh4ULVBQe1I139hLEMmfs6L3Kk0dSj29gLeLgIAICIiAgAgIiICACAiIgIAICIiAgAgIiICACAiIgIAICIiAgAgIiICACAiIgIAICIiAgAgIiICACAiIgIAICIiAgAgIiICACAiIgIAICIiAgAgIiICACAiIgIAICIiAgAgIiICACAiIgIAICIiAgAgIiICACAiIgIAICIiAgAgIiICACAiIgIAICIiAgAgIiICACAiIgIAICIiAgAgIiICACAiIgIAICIiAgAgIiICACAiIgIAICIiAgAgIqDEg9TN6ClDb6frpZkDoH6bmn+wevcyCXkZHPwxoHMdw7HGRA30+9vhxH8Ux1ZOADB13dN8ee1g7B7RkPp+PBxWPmYUWqZOLjeIIL9RsDRbMxwP66Hg8fc2OgMYx9yd9lxBcTDrXCwGN8WUJRUoGpO057VLcysVeC8jDGENAmhHsE2q2jM+PABr9IRrSbCT9AX10GoPeCmgcjUNFr1oAHblYu2X8BiAhg8oD2z+pXQyyP/nElV6zdMrKpYW/y17JPAYm2uYx81aNR1H6pJx5fJKvPs6AEKFOgIw3qIPzRvcTc3yA1JrQqmi231lqj1mQsJ6DpitAWjQGqbFtxaN9wtp8+Tig5TU9L+nmDZ4xAFoOyrhZ444+Y0Hl6gtofrNvQY6OT2XalkMznRTQdViuj6gAJJB2jEH2zS4g36bGht6CWv5EnASCfYnSmrDkhI8/q2MWA+cpG6EjIJ/TlkpRhGlvgle8LpPFwPhPePdYHRRjsq9/YtmotKumTaxyubA3oCRQiTNe0KwmQchU0qbd8MF8C6jHttpqf8h/+zQgF3kEIGNAY+RjE16SDcu9bb+B2Ix8J6DEu2xSM1WQMl29sCC5zV/ubQf7hNuhH9fDfuKnxwGNmUCkDRtjWtLFSiqHZPr5xKN9wpU+PnF5UPrTOc0XTjc6DxsDLGUSvQCkkhO0QAZmV7sulisc1/tDwZAetqDEy7RvNWyaVxkYKvE2dIU8IFnlxO+2m/jWnh4EFAsgQef7lXc4siyKAJkxWUBJlRO/lYCExz2e5oO+88/yooX95PEUhmngYjpnQSGkB16P3zj0DasxndH86pnV9xvz+xrTpgPevcIxoE0M8vtDjpN9eUsWc7Yzfm059A3DtlsmBS52vDm743n2JURn53A3A0ofXkj4GFk/mwF5R0uuHXBhQkcQk1ow+TK6mH37tAUFRMbPDB9TShtMZgvfDA18gsNdW0y2u09T2xULQrvNxdMlDqaNj82IzK9rQdIaKLRlDTfvo/Z21aoB2cq/6tT96cbQjNk8Zozo65vX1ZaJ8uajCsO01OldxVpA4uU6oPD0i8Hj6yAfqcfYitkzZOavvHiuOKxYaMVAYUiXATnnkr72/S7N2NffD1oOSpmU5jK/J1R3nz5fHBYvtGZgXc+IrytsyFvQV/n9D5fyZ5m8/zUWFQP3dgEwtfd3MRmsqy+2nS6thFaNhx2x4NFfC8gbTpr946QI0A0WlHtzRvFk2Yl5PK6PtVFKAIr7ZzkaR0Z9I6A2Sqfz5mNKI1dFu23pUQDyHb7YZ5SYEKBbgnQr5QHZlsN+ZYcpY0JfaUHK7wqt+pDdvDFN0wxmPt6R5lspTBf8y7uVeGrR3d9YW5DWEU9awWaD0GR+J/fu9EKbDXTDd7dz0unSTaHwMJUIRdbFtDEigUWpTCTavZwBtFiSP5b/RZoPTEfvnSa+d63Y+pFgHZzG5bGvLH5U9goHgCyYekvqA0hl3+ZHiCyWPFSuE6dxMSgpVMs7nik6Wq2b3Qwo3e6wCk8qHs0Ug7SbJwToAh+btADU38U2JePhrNiC7HH9XyWsxtkgLRsN17emfLJXm/xhmvRLXKwiBnk+mRCUyKV6WetscOSzwmSNxpnOSwCVZ7GYu44BaWtF6ZQukUWbnn/2c1W0pOK/xjlAarPQemXOTh44W8vVQatqOY1DJuntVz3Rki4uFA3MRsdabU8OT9v7z2lUCmlehsA0HAaTrc9VNQuFA/sAGnUKSDycl1RGogAKm2chxsxS4d2V5R0sFA/sAmhTP8eHYEcRl7JxbbKetdiNUtOpXFW+0IKBPQDJajHakXyM0Z+WiX3K2Is5GED6YUBNbr3sni0A+aiUxO1MbrcWZB1MTUKXVogWenrgxelGGY3GtQVlTjROZ21IT9OGT8j6J3G9DVDMZWPOxXKAtOGjrZcp17frtOw5b0wvAxStRwbsVQO7nshDMS6mPKDI5FLWrwWkRMN5/qp71w19a9gm2jb46/bUhR2R3WNPYVXfocKFXhl4YbosEnjuMFuQHgZrOGoSvmU1JS9NFnpp4PEs181vfeqMZvJ5XoctDe0/BzD3ASrY7mh4XXTuMP/MR7GlkdA616GeWehmi6dTHYS1btcHrUV+94N0yPInOtQzC02a6GxB2+e6m/NcWaiDTcyAJvcbHKtdpXgyBl2OSKcnMOErADLutcgDilVhI0K/ZkFW3mkGA0hHK3J/Oun62xc7+sEYFG9SDAHOrCGYTPSvZwA9lsVkdPb9+/wyzByGWCkmL1eWWLnQu5rVfcU7W27PZ3LpfViqRP+tb1mfB3QchdCzcid5rxp2X/+IJD/J+PMcIFQK2339gwucBeTyeOpJxpKmtfRzQRp2CsJ6mgKaQgKbtIhHS7cRb5umPetVVQMq2k4UcbwlIN89pHYiIo63rlb7rTULVbkXOOvR/1igWEndF00lMurjXJULrbIgQeg6oBBxg8G4/t28DOGhjYRX+fRAfWJQYkPJ1ye3OyZxmOueaYg4hrjPkdhQI3XKYu2uaxX3mIOGgCL1s0bBuXqhlXVQs+sarfYIh+hRQxqH/O2NdupbSe+fVDWd5GO3x5YGflrS+2C7DN/BX9t/vrTQBwFN4jiE7TGz1ToMMsHrAOnMYi8u9DlAPnst7we/A2QPditotaV46TbY+YU+AEjshmW+FApGFq2omV7czfvdsENAi00NrfafTy60cmCr6cJuWALI+5VEJG5ptF1k2UJrB7abzoBJPGztW1ZdLKdmoXUD201nLchakvmclR6O5ris1wMatCfgAs4GDwEN7tMakAtIbRe21tsBeQK57L4pAXroZkBnu/khxCNpPnfozRYUNbh0HwD1dy2v3wCUWs+2vO6o3wEUIbVdCdDPAbob0e8AWqX4u/QzgNYF0F36GUCagI41JH1828Uc6c0bZqkGUQR17r+k+gBSCpXMVwDF/vUGdbIgeIf6NCD9NwC54a0B6T8Ug8At/BMhbbvBeIc6Bun9Z19OTfeQemaxzLMvtdsd3mpuNhuhrmn+AEPZdMNq1/UBvbpQDHd+/iygyzEoe3P1VhEQ0C8A0nSxXf11QAfTlWkwh2EoHN5DDwE6Of/9o4pFQPdOd3J+AnrdqGIR0L3TnZyfgF43qlgEdO90J+f/jwH9vAgIiICACAiIgIAICIiAgAgIiICACAiIgIAICIiAgLoDKrjFUnYTpvheTdu/0h3bHeAaBUOKR+nDZ0/OqDMg9Mhe2ZDiUTr79NIl3RKDGgAqmMiPICAw4vcAHV+jAlDDSFWsvwVIFU9VrBsAFSSxsnWU8fkVQKFsaZifSq5Z82hLifoXimUDysJLmytW6QX3xcoLxWaXrFDvQrHE5kucosJ5fgrQ74uAgAgIiICACAiIgIAICIiAgAgIiICACAiIgIAICIiAgAgIiICACAiIgIAICIiAgAgIiICACAiIgIAICIiAgAgI6HWA1MEn/H17vQ1Q2UOqjR/1PbzUbVcqUxGg1o/6Hl/rXVLyOUP/QaX//8/mT7IeL+hVCiDcm+Tzatx9K3qRoqWo8D63RAIioKzkg3bRszLP3v23gOJRpdbDGGREQMdS8ZVZLKcI6LAO0v8toDeKgIAICIiAgAgIiICACAiIgIAICIiAgAgIiICACAiIgIAICIiAgAgIiICACAiIgIAICIiAgAgIiICA/gGnq1eWpcnpYAAAAABJRU5ErkJggg==" /><!-- --></p>
</div>
<div id="sex-biased-dispersal" class="section level1">
<h1>Sex-biased dispersal</h1>
<p>A new function to detect sex-biased dispersal, <code>sexbias.test</code> based on <a href="http://onlinelibrary.wiley.com/doi/10.1046/j.1365-294X.2002.01496.x/abstract">Goudet et al. (2002)</a> has been implemented. To illustrate its use, load the <em>Crocidura russula</em> data set. It consists of the genotypes and sexes of 140 shrews studied by <a href="http://rspb.royalsocietypublishing.org/content/264/1378/127">Favre et al. (1997)</a>. In this species, mark -recapture showed an excess of dispersal from females, an unusual pattern in mammals. This is confirmed using genetic data:</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">&quot;crocrussula&quot;</span>)
aic&lt;-<span class="kw">AIc</span>(crocrussula$genot)
<span class="kw">boxplot</span>(aic~crocrussula$sex)</code></pre>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAAABlBMVEUAAAD///+l2Z/dAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAEEUlEQVR4nO3c0U7cMBBAUfP/P12kLUJA1jeOZ+zYufeB7oMzTg8oyXZRy4dVK7NP4O4JBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQJBAkECQQ1Ab0Wl89SzuWWtQN9fzlasUxpQKV63DI/WoOB2r8xs/MnCEoC+vwJKdXD6uOirgsB5QB9vIwqR7WN6xPp4k4DCh0Xu/mFJ5boqdHjBBq6+YZAE/d+IlBTAkFrAE3kfCLQhtcggYZuLlDktDWAWof3de1ElwIKPFogOHoNoMbNBYpdXj36HkCdV9Xf57YhUOQwgWCYQDBMIBi244OiQDBOIBj3wGvQ4Xej/yzODFsCqPx5EXMWZ4atAFQOXwacxZlhSwP1Xglxu4Bx/gQ1TdvzGrTAbX7qXSxy+U2egyK/55knGrHwyjiBYJxAMM5rEIwTCMYJBOMEgnECwbixQDmzBRp7EgI1jfMaBOMEgnECwbix78UWBBq7+YZAExMIEghaA8hr0NDNBYqclgP0Wlx7QpkI1PbclAj0/aV3XCdQ12NmHlCpHjftLlYatxfo1PrIhV+LBaot7vy/O/pW86yGeWm3+XL4f3dcfAO+3V0setzENyYCRW196RzfHyQQHOS7eThIIDhIoNBx216DosYJNHbzlK0FilqYMU6g+yYQJBC0BpDXoLGbp2wtUNTCjHECjd08ZWvvYlEL/y+Hf5t/OlD586Jr3MTyPjisH/fwa5BA5xcLVF99v4+eG7vnXSz2F8e78jkIEggSCBoMNOUy0lXSbR4vqA8H4tVPB8Lljwei9QKNHZeXQJBAUC7Q+4Nanxqu7B7SCkDl6vYRLQBUru8fkECndo9cGD5OoDNrN70GBY3b9y42bFxe04CWaRJQW7Gb5/xVBJoydcrmAg2dljt1yuYCDZ2WO3XK5gINnZY7dcrmGwKtkECQQJBAkECQQJBAkECQQJBAkECQQJBA0LyPNhs/fqnOOngV1f0/+z0zq/x+ETg7fOKEjQXCWeXHn5EJhLMnFQv09bs0OwHF3cQ2BQqeVZJ+IUugE7OnJNDAjcvPL6EJdGa2vU8gSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCBIIEggSCDoH1tnPUO9fWz6AAAAAElFTkSuQmCC" /><!-- --></p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">tapply</span>(aic,crocrussula$sex,mean)</code></pre>
<pre><code>##          F          M 
## -1.1602396  0.9770438</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sexbias.test</span>(crocrussula$genot,crocrussula$sex)</code></pre>
<pre><code>## $call
## sexbias.test(dat = crocrussula$genot, sex = crocrussula$sex)
## 
## $statistic
##         t 
## -4.117605 
## 
## $p.value
## [1] 8.097862e-05</code></pre>
</div>
<div id="simulating-genetic-data" class="section level1">
<h1>Simulating genetic data</h1>
<p>It is now possible to simulate genetic data from a continent islands model, either at equilibrium via <code>sim.genot</code> or for a given number of generations via <code>sim.genot.t</code>. These two functions have several arguments, allowing to look at the effect of population sizes, inbreeding, migration and mutation. the number of independant loci and number of alleles per loci can also be specified. It is also possible to simulate data from a finite island model using <code>sim.genot.t</code>, by specifying that argument <code>IIM</code> is <code>FALSE</code>.</p>
</div>
<div id="exporting-to-other-programs" class="section level1">
<h1>Exporting to other programs</h1>
<p><code>hierfstat</code>can now export to files suitable for <a href="http://cmpg.unibe.ch/software/BayeScan/">Bayescan</a>, <a href="http://pngu.mgh.harvard.edu/~purcell/plink/">plink</a> and <a href="http://pritchardlab.stanford.edu/structure.html">structure</a>. Functions to import from these program can be found in <code>adegenet</code>. The functions to export to these programs are <code>write.bayescan</code>, <code>write.ped</code> and <code>write.struct</code>respectively.</p>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
