<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Jerome Goudet" />

<meta name="date" content="2016-08-23" />

<title>Importing data in Hierfstat</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; }
code > span.dt { color: #902000; }
code > span.dv { color: #40a070; }
code > span.bn { color: #40a070; }
code > span.fl { color: #40a070; }
code > span.ch { color: #4070a0; }
code > span.st { color: #4070a0; }
code > span.co { color: #60a0b0; font-style: italic; }
code > span.ot { color: #007020; }
code > span.al { color: #ff0000; font-weight: bold; }
code > span.fu { color: #06287e; }
code > span.er { color: #ff0000; font-weight: bold; }
</style>



<link href="data:text/css,body%20%7B%0A%20%20background%2Dcolor%3A%20%23fff%3B%0A%20%20margin%3A%201em%20auto%3B%0A%20%20max%2Dwidth%3A%20700px%3B%0A%20%20overflow%3A%20visible%3B%0A%20%20padding%2Dleft%3A%202em%3B%0A%20%20padding%2Dright%3A%202em%3B%0A%20%20font%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0A%20%20font%2Dsize%3A%2014px%3B%0A%20%20line%2Dheight%3A%201%2E35%3B%0A%7D%0A%0A%23header%20%7B%0A%20%20text%2Dalign%3A%20center%3B%0A%7D%0A%0A%23TOC%20%7B%0A%20%20clear%3A%20both%3B%0A%20%20margin%3A%200%200%2010px%2010px%3B%0A%20%20padding%3A%204px%3B%0A%20%20width%3A%20400px%3B%0A%20%20border%3A%201px%20solid%20%23CCCCCC%3B%0A%20%20border%2Dradius%3A%205px%3B%0A%0A%20%20background%2Dcolor%3A%20%23f6f6f6%3B%0A%20%20font%2Dsize%3A%2013px%3B%0A%20%20line%2Dheight%3A%201%2E3%3B%0A%7D%0A%20%20%23TOC%20%2Etoctitle%20%7B%0A%20%20%20%20font%2Dweight%3A%20bold%3B%0A%20%20%20%20font%2Dsize%3A%2015px%3B%0A%20%20%20%20margin%2Dleft%3A%205px%3B%0A%20%20%7D%0A%0A%20%20%23TOC%20ul%20%7B%0A%20%20%20%20padding%2Dleft%3A%2040px%3B%0A%20%20%20%20margin%2Dleft%3A%20%2D1%2E5em%3B%0A%20%20%20%20margin%2Dtop%3A%205px%3B%0A%20%20%20%20margin%2Dbottom%3A%205px%3B%0A%20%20%7D%0A%20%20%23TOC%20ul%20ul%20%7B%0A%20%20%20%20margin%2Dleft%3A%20%2D2em%3B%0A%20%20%7D%0A%20%20%23TOC%20li%20%7B%0A%20%20%20%20line%2Dheight%3A%2016px%3B%0A%20%20%7D%0A%0Atable%20%7B%0A%20%20margin%3A%201em%20auto%3B%0A%20%20border%2Dwidth%3A%201px%3B%0A%20%20border%2Dcolor%3A%20%23DDDDDD%3B%0A%20%20border%2Dstyle%3A%20outset%3B%0A%20%20border%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0A%20%20border%2Dwidth%3A%202px%3B%0A%20%20padding%3A%205px%3B%0A%20%20border%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0A%20%20border%2Dwidth%3A%201px%3B%0A%20%20border%2Dstyle%3A%20inset%3B%0A%20%20line%2Dheight%3A%2018px%3B%0A%20%20padding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0A%20%20border%2Dleft%2Dstyle%3A%20none%3B%0A%20%20border%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0A%20%20background%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0A%0Ap%20%7B%0A%20%20margin%3A%200%2E5em%200%3B%0A%7D%0A%0Ablockquote%20%7B%0A%20%20background%2Dcolor%3A%20%23f6f6f6%3B%0A%20%20padding%3A%200%2E25em%200%2E75em%3B%0A%7D%0A%0Ahr%20%7B%0A%20%20border%2Dstyle%3A%20solid%3B%0A%20%20border%3A%20none%3B%0A%20%20border%2Dtop%3A%201px%20solid%20%23777%3B%0A%20%20margin%3A%2028px%200%3B%0A%7D%0A%0Adl%20%7B%0A%20%20margin%2Dleft%3A%200%3B%0A%7D%0A%20%20dl%20dd%20%7B%0A%20%20%20%20margin%2Dbottom%3A%2013px%3B%0A%20%20%20%20margin%2Dleft%3A%2013px%3B%0A%20%20%7D%0A%20%20dl%20dt%20%7B%0A%20%20%20%20font%2Dweight%3A%20bold%3B%0A%20%20%7D%0A%0Aul%20%7B%0A%20%20margin%2Dtop%3A%200%3B%0A%7D%0A%20%20ul%20li%20%7B%0A%20%20%20%20list%2Dstyle%3A%20circle%20outside%3B%0A%20%20%7D%0A%20%20ul%20ul%20%7B%0A%20%20%20%20margin%2Dbottom%3A%200%3B%0A%20%20%7D%0A%0Apre%2C%20code%20%7B%0A%20%20background%2Dcolor%3A%20%23f7f7f7%3B%0A%20%20border%2Dradius%3A%203px%3B%0A%20%20color%3A%20%23333%3B%0A%20%20white%2Dspace%3A%20pre%2Dwrap%3B%20%20%20%20%2F%2A%20Wrap%20long%20lines%20%2A%2F%0A%7D%0Apre%20%7B%0A%20%20border%2Dradius%3A%203px%3B%0A%20%20margin%3A%205px%200px%2010px%200px%3B%0A%20%20padding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0A%20%20background%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0A%0Acode%20%7B%0A%20%20font%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0A%20%20font%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0A%20%20padding%3A%202px%200px%3B%0A%7D%0A%0Adiv%2Efigure%20%7B%0A%20%20text%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0A%20%20background%2Dcolor%3A%20%23FFFFFF%3B%0A%20%20padding%3A%202px%3B%0A%20%20border%3A%201px%20solid%20%23DDDDDD%3B%0A%20%20border%2Dradius%3A%203px%3B%0A%20%20border%3A%201px%20solid%20%23CCCCCC%3B%0A%20%20margin%3A%200%205px%3B%0A%7D%0A%0Ah1%20%7B%0A%20%20margin%2Dtop%3A%200%3B%0A%20%20font%2Dsize%3A%2035px%3B%0A%20%20line%2Dheight%3A%2040px%3B%0A%7D%0A%0Ah2%20%7B%0A%20%20border%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0A%20%20padding%2Dtop%3A%2010px%3B%0A%20%20padding%2Dbottom%3A%202px%3B%0A%20%20font%2Dsize%3A%20145%25%3B%0A%7D%0A%0Ah3%20%7B%0A%20%20border%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0A%20%20padding%2Dtop%3A%2010px%3B%0A%20%20font%2Dsize%3A%20120%25%3B%0A%7D%0A%0Ah4%20%7B%0A%20%20border%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0A%20%20margin%2Dleft%3A%208px%3B%0A%20%20font%2Dsize%3A%20105%25%3B%0A%7D%0A%0Ah5%2C%20h6%20%7B%0A%20%20border%2Dbottom%3A%201px%20solid%20%23ccc%3B%0A%20%20font%2Dsize%3A%20105%25%3B%0A%7D%0A%0Aa%20%7B%0A%20%20color%3A%20%230033dd%3B%0A%20%20text%2Ddecoration%3A%20none%3B%0A%7D%0A%20%20a%3Ahover%20%7B%0A%20%20%20%20color%3A%20%236666ff%3B%20%7D%0A%20%20a%3Avisited%20%7B%0A%20%20%20%20color%3A%20%23800080%3B%20%7D%0A%20%20a%3Avisited%3Ahover%20%7B%0A%20%20%20%20color%3A%20%23BB00BB%3B%20%7D%0A%20%20a%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0A%20%20%20%20text%2Ddecoration%3A%20underline%3B%20%7D%0A%20%20a%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0A%20%20%20%20text%2Ddecoration%3A%20underline%3B%20%7D%0A%0A%2F%2A%20Class%20described%20in%20https%3A%2F%2Fbenjeffrey%2Ecom%2Fposts%2Fpandoc%2Dsyntax%2Dhighlighting%2Dcss%0A%20%20%20Colours%20from%20https%3A%2F%2Fgist%2Egithub%2Ecom%2Frobsimmons%2F1172277%20%2A%2F%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%2F%2A%20Keyword%20%2A%2F%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%2F%2A%20DataType%20%2A%2F%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%2F%2A%20DecVal%20%28decimal%20values%29%20%2A%2F%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%2F%2A%20BaseN%20%2A%2F%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%2F%2A%20Float%20%2A%2F%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%2F%2A%20Char%20%2A%2F%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%2F%2A%20String%20%2A%2F%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%2F%2A%20Comment%20%2A%2F%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%2F%2A%20OtherToken%20%2A%2F%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%2F%2A%20AlertToken%20%2A%2F%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%2F%2A%20Function%20calls%20%2A%2F%20%0Acode%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%2F%2A%20ErrorTok%20%2A%2F%0A%0A" rel="stylesheet" type="text/css" />

</head>

<body>




<h1 class="title toc-ignore">Importing data in Hierfstat</h1>
<h4 class="author"><em>Jerome Goudet</em></h4>
<h4 class="date"><em>2016-08-23</em></h4>



<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>This vignette documents how to import or enter genotypic data for the <code>hierfstat</code> package. Originally this package was written to estimate and test hierarchical F-statistics, but was then further developped and now include almost all features of the <a href="http://www2.unil.ch/popgen/softwares/fstat.htm">Fstat program</a>, which is no longer maintained, as well as several others.</p>
</div>
<div id="format-required-by-most-functions-in-hierfstat" class="section level1">
<h1>Format required by most functions in <em>Hierfstat</em></h1>
<p>The data types that <code>hierfstat</code> can analyse are haploid or diploid, unphased, multilocus genotypes. Note that each data set must be made of only one ploidy level.</p>
<p>The basic data structure required by most <code>Hierfstat</code> function is a data frame with the first column containing a population identifier (preferably a number), and the next <span class="math">\(nl\)</span> columns the genotype at each of <span class="math">\(nl\)</span> loci.</p>
<p>In <code>hierfstat</code>, alleles are encoded as 1, 2 or 3 digits numbers, and genotypes are encoded as numbers with the two alleles collated (as in pasted together). Other type of data can be imported (see below) but for the time being we focus on the primary data type. Thus imagine that you have an individual genotyped at a microsatellite locus with allele length <code>120</code> and <code>124</code>, the way to encode it for <code>hierfstat</code> is either <code>120124</code>or <code>124120</code>. If the data are SNPs, each allele at a locus could be encoded as <code>1</code> and <code>2</code>, or you may decide to keep the correspondence between nucleotides and alleles (e.g. <code>1, 2, 3, 4</code> for <code>A, C, G, T</code>). Thus, if the two alleles at a SNP locus are <code>A</code> and <code>T</code> and an individual is heterozygote, it could be encoded as <code>14</code>or <code>41</code>.</p>
<p>Example data sets are included in <code>hierfstat</code>. For instance:</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(hierfstat) <span class="co">#load the library</span>
<span class="kw">data</span>(diploid) <span class="co"># info about this data set with ?diploid</span>
<span class="kw">head</span>(diploid)</code></pre>
<pre><code>##   Pop loc-1 loc-2 loc-3 loc-4 loc-5
## 1   1    44    43    43    33    44
## 2   1    44    44    43    33    44
## 3   1    44    44    43    43    44
## 4   1    44    44    NA    33    44
## 5   1    44    44    24    34    44
## 6   1    44    44    NA    43    44</code></pre>
<p>The first individual (first row of the diploid data frame) belongs to population 1. Its genotype at <code>loc-1</code> is <code>44</code>, thus homozygote for allele <code>4</code>. It is heterozygote for alleles <code>3</code>and <code>4</code> at both <code>loc-2</code> and <code>loc-3</code>, and homozygote for allele <code>3</code> at <code>loc-4</code> and finaly homozygote for allele <code>4</code> at <code>loc-5</code>. In fact, <code>loc-1</code> and <code>loc-4</code> are monomorphic, meaning that only one allele is present in all individuals from all populations.</p>
<p>If a genotype is missing, it is encoded as <code>NA</code>. For instance, the fourth individual has not been typed at <code>loc-3</code>, nor did the 6th individual for the same locus.</p>
<p>The first column of this dataframe contains the identifier of the population to which the individual belongs. We can find how many individuals were typed in each population by using teh command table:</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">table</span>(diploid[,<span class="dv">1</span>])</code></pre>
<pre><code>## 
## 1 2 3 4 5 6 
## 8 8 5 7 9 7</code></pre>
<p>As another example, we look at dataset <code>cont.isl99</code>, a data frame where alleles are encoded as 2 digits numbers:</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(contisl99)
<span class="kw">head</span>(cont.isl99)</code></pre>
<pre><code>##   Pop loc.1 loc.2 loc.3 loc.4 loc.5
## 1   1  7474  1955  9168  4051  9251
## 2   1  7474  3175  9168  2410  2327
## 3   1   808  3194  9536  9751  9223
## 4   1   874  5294  1876  1310  1292
## 5   1  7484  3875  1010  5107  7712
## 6   1   874  3175  1010  5135  9292</code></pre>
<p>the first individual is homozygous for allele <code>74</code> at the first locus (<code>loc.1</code>) and heterozygous fore alleles <code>19</code> and <code>55</code> at the second locus. The genotype could have been written <code>5519</code> instead of <code>1955</code>, it does not matter. Note the genotype of the 3rd and fourth individual at the first locus. They both carry allele <code>8</code>, which is in fact encoded as <code>08</code>. When it comes first, the leading 0 disappears, but it must be present in second position. Hence genotype <code>874</code>, <code>0874</code> and <code>7408</code> are the same, but different from genotype <code>748</code> who would be understood by hierfstat as an individual heterozygous for alleles <code>07</code> and <code>48</code>.</p>
<p>Last point: alleles for all loci to be analysed simultaneously must be encoded with the same number of digits.</p>
</div>
<div id="importing-data-files" class="section level1">
<h1>Importing data files</h1>
<p>Often the data to be imported are in a text file. If this is the case, the easiest way to import the file into <code>R</code> is via one of the workhorse of R, the <code>read.table</code> function.</p>
<div id="importing-fstat-data-files" class="section level2">
<h2>Importing FSTAT data files</h2>
<p>If the data are in the <code>FSTAT</code> format, they can be readily imported using the function <code>read.fstat</code>:</p>
<pre class="sourceCode r"><code class="sourceCode r">dip&lt;-<span class="kw">read.fstat</span>(<span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>,<span class="st">&quot;diploid.dat&quot;</span>,<span class="dt">package=</span><span class="st">&quot;hierfstat&quot;</span>))
<span class="kw">head</span>(dip)</code></pre>
<pre><code>##   Pop loc-1 loc-2 loc-3 loc-4 loc-5
## 1   1    44    43    43    33    44
## 2   1    44    44    43    33    44
## 3   1    44    44    43    43    44
## 4   1    44    44    NA    33    44
## 5   1    44    44    24    34    44
## 6   1    44    44    NA    43    44</code></pre>
</div>
<div id="importing-from-adegenet-genind-objects" class="section level2">
<h2>Importing from adegenet: genind objects</h2>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(adegenet)</code></pre>
<pre><code>## Loading required package: ade4</code></pre>
<pre><code>## 
##    /// adegenet 2.0.1 is loaded ////////////
## 
##    &gt; overview: '?adegenet'
##    &gt; tutorials/doc/questions: 'adegenetWeb()' 
##    &gt; bug reports/feature requests: adegenetIssues()</code></pre>
<pre><code>## 
## Attaching package: 'adegenet'</code></pre>
<pre><code>## The following objects are masked from 'package:hierfstat':
## 
##     .genlab, read.fstat</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(nancycats)
<span class="kw">head</span>(<span class="kw">genind2hierfstat</span>(nancycats)[,<span class="dv">1</span>:<span class="dv">10</span>]) <span class="co"># only the first 10 loci</span></code></pre>
<pre><code>##      pop   fca8  fca23  fca43  fca45  fca77  fca78  fca90  fca96  fca37
## N215 P01     NA 136146 139139 116120 156156 142148 199199 113113 208208
## N216 P01     NA 146146 139145 120126 156156 142148 185199 113113 208208
## N217 P01 135143 136146 141141 116116 152156 142142 197197 113113 210210
## N218 P01 133135 138138 139141 116126 150150 142148 199199  91105 208208
## N219 P01 133135 140146 141145 126126 152152 142148 193199 113113 208208
## N220 P01 135143 136146 145149 120126 150156 148148 193195  91113 208208</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="co">#basic.stats(nancycats)</span>
<span class="co">#genet.dist(nancycats)</span>
<span class="kw">data</span>(H3N2)
<span class="kw">head</span>(<span class="kw">genind2hierfstat</span>(H3N2,<span class="dt">pop=</span><span class="kw">rep</span>(<span class="dv">1</span>,<span class="kw">dim</span>(H3N2@tab)[<span class="dv">1</span>]))[,<span class="dv">1</span>:<span class="dv">10</span>]) <span class="co"># only the first 10 positions</span></code></pre>
<pre><code>##          pop X6 X17 X39 X42 X45 X51 X60 X72 X73
## AB434107   1  1   1   3   2   3   2   3   3   2
## AB434108   1  1   1   3   2   3   2   3   3   2
## AB438242   1 NA  NA  NA  NA  NA  NA   3   3   2
## AB438243   1 NA  NA  NA  NA  NA  NA   3   3   2
## AB438244   1 NA  NA  NA  NA  NA  NA   3   3   2
## AB438245   1 NA  NA  NA  NA  NA  NA   3   3   2</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="co">#basic.stats(genind2hierfstat(H3N2,pop=rep(1,dim(H3N2@tab)[1])),diploid=FALSE)</span></code></pre>
</div>
<div id="importing-from-quantinemo" class="section level2">
<h2>Importing from Quantinemo</h2>
<p><a href="http://www2.unil.ch/popgen/softwares/quantinemo/tutorial/">Quantinemo</a> is a genetic simulation program for markers and traits. Data generated by <code>Quantinemo</code>can be imported using teh function <code>qn2.read.fstat</code>. The component <code>$dat</code> of the object return by this function contains the genotypes of the individuals simulated:</p>
<pre class="sourceCode r"><code class="sourceCode r">dat&lt;-<span class="kw">qn2.read.fstat</span>(<span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>,<span class="st">&quot;qn2_sex.dat&quot;</span>,<span class="dt">package=</span><span class="st">&quot;hierfstat&quot;</span>))
<span class="kw">names</span>(dat)</code></pre>
<pre><code>## [1] &quot;dat&quot; &quot;sex&quot;</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(dat$sex)</code></pre>
<pre><code>## [1] &quot;F&quot; &quot;F&quot; &quot;F&quot; &quot;F&quot; &quot;F&quot; &quot;F&quot;</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(dat$dat[,<span class="dv">1</span>:<span class="dv">10</span>])</code></pre>
<pre><code>##   Pop trait-1_locus-1 trait-1_locus-2 trait-1_locus-3 trait-1_locus-4
## 1   1             606            1515             101             404
## 2   1             606            1515             101             404
## 3   1             606            1515             101             404
## 4   1             606            1515             101             404
## 5   1             606            1515             101             404
## 6   1             606            1515             101             404
##   trait-1_locus-5 trait-1_locus-6 trait-1_locus-7 trait-1_locus-8
## 1             707             404             303             101
## 2             707             415             303             101
## 3             707             404             303             101
## 4             707             415             303             101
## 5             707             415             303             101
## 6             707            1504             303             101
##   trait-1_locus-9
## 1             808
## 2             808
## 3             808
## 4             808
## 5             808
## 6             808</code></pre>
<pre class="sourceCode r"><code class="sourceCode r"><span class="co">#sexbias.test(dat[[1]],sex=dat[[2]])</span></code></pre>
</div>
<div id="importing-from-ms" class="section level2">
<h2>Importing from ms</h2>
<p>The program <a href="http://home.uchicago.edu/rhudson1/source/mksamples.html">ms</a> of Hudson is commonly used to generate genomic data.</p>
<p>I briefly discussed the <code>ms</code> software. its output looks like this:</p>

<p>The first line is the <code>ms</code> command line, and it instructed the program to simulate 2 populations, with <span class="math">\(\theta=2N_0\mu=20\)</span>. The 2 populations differ in size and the smallest (the second) isa 100th of the first. The two populations exchange <span class="math">\(4Nm=40\)</span> migrants per generation. 100 chromosomes is sampled from each population, and this is repeated a 100 times.</p>
<p>The genetic data itself comes as a series of 0 and 1, collated one to the other. These are the SNPs site, with 0 being the ancestral state and 1 the mutatnt state.</p>
<p>the function <code>read.ms.snps</code> allows reading this into R:</p>
<pre class="sourceCode r"><code class="sourceCode r">msdat&lt;-<span class="kw">read.ms.snps</span>()</code></pre>
</div>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
