#################################
#' @title Read data generated by Hudson ms program
#'  
#' @description Read data generated by \href{http://home.uchicago.edu/rhudson1/source/mksamples.html}{Hudson ms} program 
#' 
#'  
#' @usage ms.read.snps(fname)
#' @param fname ms data file name
#' @return alldat  a data frame with nloc+1 columns, the first being the population
#'  to which the individual belongs and the next being the genotypes, one column per locus; 
#'  and ninds rows
#' @details Each site is read as a SNP, with the ancestral allele encoded as 0 and the alternate 
#' allele encoded as 1.  If the ms output file contains several replicates, 
#' the different replicates will be collated together.
#' @author Jerome Goudet \email{jerome.goudet@@unil.ch}
#' @references \href{http://bioinformatics.oxfordjournals.org/content/18/2/337.short}{Hudson, R. R. (2002) Generating samples under a Wright-Fisher neutral model of genetic variation}. Bioinformatics 18 : 337-338.
#'
#' @examples 
#' \dontrun{
#'   dat<-read.ms.snps(system.file("extdata","2pops_asspop.txt",package="hierfstat"))
#'   betas(dat)$betaiovl # population specif FSTs
#'   }
#' @export
#####################################################################################
read.ms.snps<-function(fname){
  dat<-readLines(fname)
  tmp<-strsplit(dat[1],split=" ")[[1]]
  pos.i<-which(tmp=="-I")
  if (length(pos.i)>0) {
    np<-as.integer(tmp[pos.i+1])
    ni<-as.integer(tmp[(pos.i+2):(pos.i+1+np)])
  }
  else ni<-as.integer(tmp[2])
  nrep<-as.integer(tmp[3])
  #get nsnps
  nsnps<-as.integer(unlist(lapply(strsplit(grep("segsites:",dat,value=TRUE),split=":"),function(x) x[[2]])))
  dat1<-grep("^[0,1]",dat[-c(1:4)],value=TRUE)
  nt<-sum(ni)
  dats<-split(dat1,rep(1:nrep,each=nt))
  dats1<-lapply(dats,strsplit,"")
  dats2<-lapply(dats1,function(x) matrix(as.integer(unlist(x)),nrow=nt,byrow=TRUE))
  alldat<-matrix(numeric(nt*sum(nsnps)),nrow=nt)
  cnsnps<-cumsum(nsnps)
  alldat[,1:cnsnps[1]]<-dats2[[1]]
  for (i in 2:nrep)
    alldat[,(cnsnps[i-1]+1):cnsnps[i]]<-dats2[[i]]
  popid<-rep(1:length(ni),ni)
  alldat.names<-c("Pop",paste("loc",1:dim(alldat)[2],sep="."))
  alldat<-data.frame(Pop=popid,alldat)
  names(alldat)<-alldat.names
  return(alldat)
}
